src/project_types/sequence.rs

```diff
@@ -25,42 +30,180 @@ impl Sequence {
+    /// Also writes it to file, so it can be read later
+    fn create_default_section(
+        &self,
+        num_channels: u32,
+        num_frames: u32
+    ) -> Result<SequenceSection, Error> {
+        self.create_section(
+            1,
+            num_frames,
+            vec![vec![0; num_frames as usize]; num_channels as usize]
+        )
+    }
+
+    /// Creates a sequence section, writes it to a file, and returns it
+    fn create_section(
+        &self,
+        index: u32,
+        num_frames: u32,
+        data: Vec<Vec<u8>>
+    ) -> Result<SequenceSection, Error> {
+        let section_path = self.get_section_path(index);
+        let section = SequenceSection {
+            seq_name: self.name.to_string(),
+            index: index,
+            path: section_path.clone(),
+            num_frames: num_frames,
+            data: data
+        };
+        let _ = try!(section.write_to_file());
+        Ok(section)
+    }
+
+    /// Sets the data for a sequence section
+    /// Writes the changes and commits
+    pub fn set_section_data(&self, index: u32, data: Vec<Vec<u8>>) -> Result<(), Error> {
+        let mut seq_sec = try!(self.get_section(index));
+        seq_sec.data = data;
+        let _ = try!(seq_sec.write_to_file());
+
+        let signature = Signature::now("Proton Lights", "proton@teslaworks.net").unwrap();
+        let msg = format!("Setting section data for sequence '{}', section {}", self.name, index);
+        let repo_path: Option<&Path> = None;
+
+        utils::commit_all(repo_path, &signature, &msg)
+            .map(|_| ())
+    }
+
+    /// Resection a sequence
+    pub fn resection(&mut self, num_sections: u32) -> Result<(), Error> {
+        // No need to resection if same number of sections
+        if self.num_sections == num_sections {
+            return Ok(());
+        }
+        // Cannot resection into 0 sections
+        if num_sections == 0 {
+            return Err(Error::InvalidSequenceSection(num_sections));
+        }
+
+        let num_channels: u32 = 3; // TODO change when add layout
+        let music_duration_ms = self.music_duration_sec as f32 * 1000_f32;
+        let num_frames_f32: f32 = music_duration_ms as f32 / self.frame_duration_ms as f32;
+        let num_frames = num_frames_f32.ceil() as u32;
+        let num_frames_per_section_f32 = num_frames_f32 / num_sections as f32;
+        let num_frames_per_section = num_frames_per_section_f32.ceil() as u32;
+
+        // Turn into one section if not already, then split up
+        let mut sections = try!(self.get_all_sections());
+        // If no sections (first run?), initialize with default
+        if sections.len() < 1 {
+            let default_section = try!(self.create_default_section(num_channels, num_frames));
+            sections = vec![default_section];
+        }
+        // Function to combine data vectors in the fold
+        fn combine_data(mut accumulated: Vec<Vec<u8>>, sec: &SequenceSection) -> Vec<Vec<u8>> {
```

Comment:
Since it was only used in this function, essentially as a lambda, I went Pythonic and kept it in the function. Is that done differently in Rust then?